tower-buffer/src/worker.rs

```diff
@@ -157,30 +165,43 @@ where
 
         loop {
             match try_ready!(self.poll_next_msg()) {
-                Some(msg) => {
+                Some((msg, first)) => {
+                    let _guard = msg.span.enter();
                     if let Some(ref failed) = self.failed {
+                        tracing::trace!("notifying about worker failure");
                         let _ = msg.tx.send(Err(failed.clone()));
                         continue;
                     }
 
                     // Wait for the service to be ready
+                    tracing::trace!(
+                        { resumed = !first },
+                        "worker received request; waiting for service readiness"
```

Comment:
Nit/TIOLI: IMO this might read better as 
```rust
                    tracing::trace!(
                        resumed = !first,
                        message = "worker received request; waiting for service readiness",
                    );
```
but, up to you.

Reply:
My one concern with this change is that it hardcodes an assumption that the string message is equivalent to a field named `message`. Not sure that that's something `tracing` guarantees?

Reply:
> My one concern with this change is that it hardcodes an assumption that the string message is equivalent to a field named `message`. Not sure that that's something `tracing` guarantees?

It should be guaranteed to have the same behaviour as using the `log`-like syntax; the macros will desugar that to a field named `message` with a `format_args!` value identically to the `key=value` syntax above.

Reply:
Is that documented anywhere?

Reply:
if you weren't aware of it, it is probably under-documented >.<

Reply:
So, to clarify, I was aware that the current implementation exposes it as a field called `message`. I did not know that `tracing` _guarantees_ that this will be the case (i.e., it's a part of the library's stable API).

Reply:
@hawkw I think I'm only waiting on an answer to ^ before this is ready to merge.

Reply:
I think we've committed to `message` being part of the stable API at this point --- if we ever change the handling of textual messages, we'll ensure that it is at least back-compatible with subscribers that expect a message field. This should, perhaps, be documented in `tracing`.